src/app/clusters/resource-monitoring-server/resource-monitoring-server.cpp

```diff
@@ -389,10 +349,67 @@ void Instance::HandleResetCondition(HandlerContext & ctx,
                                     const ResourceMonitoring::Commands::ResetCondition::DecodableType & commandData)
 {
 
-    Status resetConditionStatus = OnResetCondition();
+    Status resetConditionStatus = mDelegate->OnResetCondition();
     ctx.mCommandHandler.AddStatus(ctx.mRequestPath, resetConditionStatus);
 }
 
+Status Delegate::OnResetCondition()
```

Comment:
Is there a reason why this was moved from the Instance to the Delegate? `mInstance` is used a lot, it makes me think that in should be with the Instance.

Reply:
This method may be overwritten by the SDK user if the default behavior is not desired.
So this would miss the whole point of this PR if left in the instance right?

Reply:
OK. I was under the impression that the method implements behaviour that is defined in the spec, meaning that an SDK user would not want to change it's behaviour to ensure that they are aligned.